### ISSUE TITLE
Inaccuracies in Proposal History

### ISSUE THREAD

**Issue by InternetComputerOG** [peterpeterparker/proposals.network#37]
<img width="1189" alt="image" src="https://github.com/peterpeterparker/proposals.network/assets/85028872/94bd7140-18e6-4649-9751-02c12b9d0c12">

I submitted Proposal #32 for Neutrinite a while back, so when I logged in I was surprised to see the logo for ICLighthouse for this history entry.

Clicking either of the Link icons leads you to a screen with a link to open your proposal. Once you click "Open" you see the correct proposal for Neutrinite.

Clocking the "View" button takes you to that same screen, except this time clicking "Open" will take you to Proposal #32 for ICLighthouse, which is incorrect since I never submitted that one.

Basically, it's just a minor inaccuracy in the proposal history, and what should be the same path is taking users to different places depending on which element they click first.

**Comment by peterpeterparker** [comment]
Somehow governance ID (field `description`) inherited the incorrect governance ID. Should check the wizard after proposals are submitted.

<img width="1536" alt="Capture d’écran 2024-04-16 à 22 19 54" src="https://github.com/peterpeterparker/proposals.network/assets/16886711/1031a783-4abb-40fc-aceb-2d051f160d07">

**Comment by peterpeterparker** [comment]
I cannot replicate the issue. However, I blindy implemented a potential improvement for an edge case scenario.
Please let me know if this happens in the future again. If so, let's reopen the issue.
Thanks for reporting it.
